src/edid.c

```diff
@@ -95,14 +382,79 @@ static void decode_display_descriptor(const unsigned char *desc, struct monitor_
 	}
 }
 
-static void decode_descriptors(const unsigned char *edid, struct monitor_info *info)
+static void decode_detailed_timing(const unsigned char *timing, struct detailed_timing * detailed)
 {
-	for (int i = 0; i < 4; ++i) {
+	int bits;
+
+	enum stereo_type stereo[] = {
+		NO_STEREO, NO_STEREO, FIELD_RIGHT, FIELD_LEFT,
+		TWO_WAY_RIGHT_ON_EVEN, TWO_WAY_LEFT_ON_EVEN,
+		FOUR_WAY_INTERLEAVED, SIDE_BY_SIDE
+	};
+
+	detailed->pixel_clock = (timing[0x00] | timing[0x01] << 8) * 10000;
```

Comment:
This code is a bit hard to read, if you add some spaces you can align all columns more and make it easier to spot bugs. E.g., align all instances of timing[...] so subsequent rows have aligned columns.  (Not important, just a matter of style)